cloud/deployment/fromfile/fromfile.go

```diff
@@ -139,16 +125,20 @@ func CreateOrUpdate(inputFile, action string, client astro.Client, coreClient as
 		}
 		// this deployment does not exist so create it
 		// transform formattedDeployment to DeploymentCreateInput
-		createInput, _, err = getCreateOrUpdateInput(&formattedDeployment, clusterID, workspaceID, createAction, &astro.Deployment{}, nodePools, dagDeploy, client)
+		err = createOrUpdateDeployment(&formattedDeployment, clusterID, workspaceID, createAction, &astroplatformcore.Deployment{}, nodePools, dagDeploy, envVars, astroPlatformCore)
 		if err != nil {
 			return err
 		}
-		// create the deployment
-		createdOrUpdatedDeployment, err = client.CreateDeployment(&createInput)
-		if err != nil {
-			return fmt.Errorf("%s: %w %+v", err.Error(), errCreateFailed, createInput)
+		if hasEnvVarsOrAlertEmails(&formattedDeployment) {
+			action = updateAction
 		}
```

Comment:
can you add a comment about why we have to do it like this

Reply:
added comment "users can not creat a deployment with env vars or alert emails in v1beta1 so to create them with a deployment file we need update right after deployment creation"